data-prepper-plugins/http-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/service/HttpSinkService.java

```diff
@@ -205,32 +297,54 @@ private HttpAuthOptions getAuthHandlerByConfig(final AuthTypeOptions authType,
      *  @param httpSinkConfiguration HttpSinkConfiguration.
      */
     private Map<String,HttpAuthOptions> buildAuthHttpSinkObjectsByConfig(final HttpSinkConfiguration httpSinkConfiguration){
-        final List<UrlConfigurationOption> urlConfigurationOptions = httpSinkConfiguration.getUrlConfigurationOptions();
-        final Map<String,HttpAuthOptions> authMap = new HashMap<>(urlConfigurationOptions.size());
-        urlConfigurationOptions.forEach( urlOption -> {
-            final HTTPMethodOptions httpMethod = Objects.nonNull(urlOption.getHttpMethod()) ? urlOption.getHttpMethod() : httpSinkConfiguration.getHttpMethod();
-            final AuthTypeOptions authType = Objects.nonNull(urlOption.getAuthType()) ? urlOption.getAuthType() : httpSinkConfiguration.getAuthType();
-            final String proxyUrlString =  Objects.nonNull(urlOption.getProxy()) ? urlOption.getProxy() : httpSinkConfiguration.getProxy();
-            final ClassicRequestBuilder classicRequestBuilder = buildRequestByHTTPMethodType(httpMethod).setUri(urlOption.getUrl());
-
-            if(Objects.nonNull(httpSinkConfiguration.getCustomHeaderOptions())){
-                //TODO:add custom headers
-            }
-            if(Objects.nonNull(proxyUrlString)) {
-                httpClientBuilder.setProxy(HttpSinkUtil.getHttpHostByURL(HttpSinkUtil.getURLByUrlString(proxyUrlString)));
-                LOG.info("Sending data via proxy {}",proxyUrlString);
-            }
-
-            final HttpAuthOptions.Builder authOptions = new HttpAuthOptions.Builder()
-                    .setUrl(urlOption.getUrl())
-                    .setClassicHttpRequestBuilder(classicRequestBuilder)
-                    .setHttpClientBuilder(httpClientBuilder);
-
-            authMap.put(urlOption.getUrl(),getAuthHandlerByConfig(authType,authOptions));
-        });
+        final Map<String,HttpAuthOptions> authMap = new HashMap<>();
+
+        final HTTPMethodOptions httpMethod = httpSinkConfiguration.getHttpMethod();
+        final AuthTypeOptions authType =  httpSinkConfiguration.getAuthType();
+        final String proxyUrlString =  httpSinkConfiguration.getProxy();
+        final ClassicRequestBuilder classicRequestBuilder = buildRequestByHTTPMethodType(httpMethod).setUri(httpSinkConfiguration.getUrl());
+
+        if(httpSinkConfiguration.isAwsSigv4() && httpSinkConfiguration.isValidAWSUrl()){
```

Comment:
Let's split all of this AWS code into its own class. It is too tightly coupling the code.

Reply:
Thanks for the review @dlvenable , the AWS code is moved to a new class.